packages/database/src/core/RepoManager.ts

```diff
@@ -116,6 +123,11 @@ export class RepoManager {
       dbUrl = `http://${dbEmulatorHost}?ns=${repoInfo.namespace}`;
```

Comment:
Does this branch `if (dbEmulatorHost) { }` trigger only if the emulator connection is made via env var?  What about if the user does this:

```js
firebase.initializeApp({
  databaseURL: 'http://localhost:9000?ns=namespace'
})
``` 

You might consider using the `EmulatorAdminTokenProvider` any time `Constants.NODE_ADMIN` is true and the database is `http://` (not `https://`).  That's what we do in the Firebase CLI, it catches more cases and it also has the nice side benefit of not allowing someone's admin credentials to ever go to an insecure host.